pkg/operator/staticpod/controller/common/helpers.go

```diff
@@ -0,0 +1,42 @@
+package common
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+// UpdateStatus applies the update funcs to the oldStatus, tries to update via the client and repeats
+// up to 3 times to react to conflict errors.
+func UpdateStatus(client OperatorClient, resourceVersion string, oldStatus *operatorv1.StaticPodOperatorStatus, updateFuncs ...func(status *operatorv1.StaticPodOperatorStatus) error) (bool, error) {
+	var lastErr error
+	for retry := 0; retry < 3; retry++ {
```

Comment:
I think you're looking for retryOnConflict

Reply:
> I think you're looking for retryOnConflict

client-go/util/retry

Reply:
that doesn't play well with the get-mutate-update logic. Doesn't save a line of code.

Reply:
I'd prefer to use the project standard with fewer returns

```go
	return retry.RetryOnConflict(retry.DefaultBackoff, func() error{
		_, oldStatus, resourceVersion, err = client.Get()
		if err != nil {
			return err
		}

		newStatus := oldStatus.DeepCopy()
		for _, update := range updateFuncs {
			if err := update(newStatus); err != nil {
				return err
			}
		}

		if reflect.DeepEqual(oldStatus, newStatus) {
			return nil
		}

		err := client.UpdateStatus(resourceVersion, newStatus)
		return err
})
```